virttest/utils_libvirt/libvirt_keywrap.py

```diff
@@ -0,0 +1,45 @@
+"""
+Module to contain logic testing protected key module 'pkey'
+"""
+import logging
+import os
+
+from virttest.utils_misc import cmd_status_output
```

Comment:
I am thinking how can we prevent cirrus-loop import by avoiding one utils refer to method in other utils. 
Shall we just use process.run in avocado project?

Reply:
@chunfuwen I'm not familiar with the cirrus-loop import issue, I'm sorry. Could you show me what you mean?

If this has to be avoided, I really would want to maintain usage of `cmd_status_output` to handle running a command in guest or on host in the same way without duplicating code. From what you mention `cmd_status_output` should then maybe be moved to `avocado.utils`?

Another way I can think of is to have the test code (e.g.  in tp-libvirt) pass the method to use, s. https://github.com/smitterl/avocado-vt/blob/f3f55f362e3991b3963b30df7b3ac32116de6bce/virttest/utils_zcrypt.py#L27

What do you think?

Reply:
sorry, typo issue, should be circular-imports. 
see https://stackabuse.com/python-circular-imports/
Normally, if avocado-vt refer to some modules in avocado, there is no problem here because avocado never refer to module in avocado-vt. 
However, if avocado-vt refer to avocado-vt modules, there are potential risk here because we don't have good Hierarchical 
design at the beginning. And We met a few cases before.
I find there is one method:getstatusoutput in process.py, can you try whether it can satisfy your requirement?

Reply:
> sorry, typo issue, should be circular-imports.
> see https://stackabuse.com/python-circular-imports/
> Normally, if avocado-vt refer to some modules in avocado, there is no problem here because avocado never refer to module in avocado-vt.
> However, if avocado-vt refer to avocado-vt modules, there are potential risk here because we don't have good Hierarchical
> design at the beginning. And We met a few cases before.
> I find there is one method:getstatusoutput in process.py, can you try whether it can satisfy your requirement?

Unfortunately https://github.com/avocado-framework/avocado/blob/master/avocado/utils/process.py#L1233 is not an option because this doesn't let me execute the command in the guest.

However, I believe the import chain is not circular:
virttest.utils_libvirt.libvirt_keywrap
  -> virttest.utils_misc
    -> avocado.utils.process
    -> aexpect.client.ShellSession
and
utils_misc does not import utils_libvirt.
 
I'm not sure if I'm missing something?